packages/flutter/lib/src/foundation/_identity_map_web.dart

```diff
@@ -0,0 +1,237 @@
+    return _javaScriptSet.has(element);
+  }
+
+  @override
+  Iterator<V> get iterator => _WebSetLikeIterable<V>(_javaScriptSet).iterator;
+
+  @override
+  int get length => _javaScriptSet.size;
+
+  @override
+  V? lookup(Object? element) {
+    // Because this class only supports identity operation, if the set
+    // contains the provided element it must be exactly that element.
+    if (_javaScriptSet.has(element)) {
+      return _unsafeCast<V>(element);
+    }
+    return null;
+  }
+
+  @override
+  bool remove(Object? value) {
+    return _javaScriptSet.delete(value);
+  }
+
+  @override
+  Set<V> toSet() {
+    final _WebSetLike<V> other = _WebSetLike<V>();
+    forEach(other.add);
+    return other;
+  }
+}
+
+class _WebMapLike<K, V> extends MapBase<K, V> {
+  _WebMapLike();
+
+  final _JavaScriptMap _javascriptMap = _JavaScriptMap();
+
+  @override
+  V? operator[](Object? key) => _unsafeCast<V?>(_javascriptMap.get(key));
+
+  @override
+  void operator[]=(K key, V value) => _javascriptMap.set(key, value);
+
+  @override
+  bool get isEmpty => _javascriptMap.size == 0;
+
+  @override
+  bool get isNotEmpty => _javascriptMap.size != 0;
+
+  @override
+  V? remove(Object? key) {
+    final V? value = _unsafeCast<V?>(_javascriptMap.get(key));
+    _javascriptMap.delete(key);
+    return value;
+  }
+
+  @override
+  Iterable<K> get keys {
+    return _WebMapLikeIterable<K>(_javascriptMap);
+  }
+
+  @override
+  void clear() {
+    _javascriptMap.clear();
+  }
+}
+
+class _WebSetLikeIterable<T> extends Iterable<T> {
+  _WebSetLikeIterable(this._javascriptSet);
+
+  final _JavaScriptSet _javascriptSet;
+
+  @override
+  bool contains(Object? element) {
+    return _javascriptSet.has(element);
+  }
+
+  @override
+  int get length => _javascriptSet.size;
+
+  @override
+  Iterator<T> get iterator {
+    final _JavaScriptKeyIterator backing = _javascriptSet.keys();
+    return _WebMapLikeIterator<T>(backing);
+  }
+}
+
+class _WebMapLikeIterable<T> extends Iterable<T> {
+  _WebMapLikeIterable(this._javascriptMap);
+
+  final _JavaScriptMap _javascriptMap;
+
+  @override
+  bool contains(Object? element) {
+    return _javascriptMap.has(element);
+  }
+
+  @override
+  int get length => _javascriptMap.size;
+
+  @override
+  Iterator<T> get iterator {
+    final _JavaScriptKeyIterator backing = _javascriptMap.keys();
+    return _WebMapLikeIterator<T>(backing);
+  }
+}
+
+class _WebMapLikeIterator<T> extends Iterator<T> {
+  _WebMapLikeIterator(this._backing);
+
+  final _JavaScriptKeyIterator _backing;
+  late _JavaScriptIteratorValue _value;
+
+  @override
+  T get current => _unsafeCast<T>(_value.value);
+
+  @override
+  bool moveNext() {
+    _value = _backing.next();
+    return !_value.done;
+  }
+}
+
+// Returns [any] assuming that the return type is [T].
+@pragma('dart2js:tryInline') // Required for next pragma to be effective.
+@pragma('dart2js:as:trust') // Omits `as T`.
+T _unsafeCast<T>(dynamic any) => any as T;
```

Comment:
Without unsafe casting applied here this regresses performance compared to a "regular" identity map